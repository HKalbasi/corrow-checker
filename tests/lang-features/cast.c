#include <stddef.h>

void *owned malloc(unsigned long size);
void free(void *owned ptr);

int main()
{
    // p1 is leaked, p2 is not
    void *p1 = malloc(5);
    size_t p1_addr = (size_t)p1;
    void *p2 = malloc(5);
    size_t p2_addr = (size_t)p2;
    void *p2_again = (void*)p2_addr;
    free(p2_again);
}