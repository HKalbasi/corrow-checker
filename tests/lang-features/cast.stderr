Error: Potential leak detected
    ╭─[preproccessed_file.c:1:1]
    │
 32 │     void *p1 = malloc(5);
    │              ─────┬┬────  
    │                   ╰─────── Ownership of the value transferred to this place here
    │                    │      
    │                    ╰────── Ownership of the value transferred to this function here
 33 │     size_t p1_addr = (size_t)p1;
    │                    ──────┬─────  
    │                          ╰─────── Ownership of the value transferred to this place here
    │ 
 38 │ }
    │ ┬  
    │ ╰── Function implicitly exited without proper destruction of that value
────╯